jwql/instrument_monitors/nirspec_monitors/ta_monitors/wata_monitor.py

```diff
@@ -0,0 +1,600 @@
+            query_result = 59607.0  # a.k.a. Jan 28, 2022 == First JWST images (MIRI)
+            logging.info(('\tNo query history for {}. Beginning search date will be set to {}.'.format(self.aperture, query_result)))
+        else:
+            query_result = np.max(dates)
+
+        return query_result
+
+    @log_fail
+    @log_info
+    def run(self):
+        """The main method. See module docstrings for further details."""
+
+        logging.info('Begin logging for wata_monitor')
+
+        # define WATA variables
+        self.instrument = "nirspec"
+        self.aperture = "NRS_S1600A1_SLIT"
+
+        # Identify which database tables to use
+        self.identify_tables()
+
+        # Get the output directory and setup a directory to store the data
+        self.output_dir = os.path.join(get_config()['outputs'], 'wata_monitor')
+        ensure_dir_exists(self.output_dir)
+        # Set up directories for the copied data
+        ensure_dir_exists(os.path.join(self.output_dir, 'data'))
+        self.data_dir = os.path.join(self.output_dir,
+                                     'data/{}_{}'.format(self.instrument.lower(),
+                                                         self.aperture.lower()))
+        ensure_dir_exists(self.data_dir)
+
+        # Locate the record of most recent MAST search; use this time
+        self.query_start = self.most_recent_search()
+        self.query_end = Time.now().mjd
+        logging.info('\tQuery times: {} {}'.format(self.query_start, self.query_end))
+
+        # Query MAST using the aperture and the time of the
+        # most recent previous search as the starting time
+        new_entries = monitor_utils.mast_query_ta(self.instrument, self.aperture, self.query_start, self.query_end)
+        wata_entries = len(new_entries)
+        logging.info('\tMAST query has returned {} new WATA files for {}, {} to run the WATA monitor.'.format(wata_entries, self.instrument, self.aperture))
+
+        # Get full paths to the files
+        new_filenames = []
+        for entry_dict in new_entries:
+            filename_of_interest = entry_dict['productFilename']
+            try:
+                new_filenames.append(filesystem_path(filename_of_interest))
+            except FileNotFoundError:
+                logging.warning('\t\tUnable to locate {} in filesystem. Not including in processing.'.format(filename_of_interest))
+
+        if len(new_filenames) == 0:
+            logging.warning('\t\t ** Unable to locate any file in filesystem. Nothing to process. ** ')
+
+        # Run the monitor on any new files
+        self.script, self.div = None, None
+        monitor_run = False
+        if len(new_filenames) > 0:
+            # get the data
+            try:
+                self.wata_data = self.get_wata_data(new_filenames)
+                # make the plots
+                self.script, self.div = self.mk_plt_layout()
+                monitor_run = True
+            except:
```

Comment:
@penaguerrero do you know what exception you are trying to capture here? Is it a `FileNotFoundError`?